### PR TITLE
1523977: Document the issues with duplicate metric names

### PR DESCRIPTION
**Pull request by mdboom**
The implementation of this is in https://github.com/mozilla/probe-scraper/pull/116

### Pull Request checklist ###

- [ ] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes a changelog entry or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: docs/user/adding-new-metrics.md [hunk 1]
```diff
@@ -70,7 +70,15 @@ The `lifetime` parameter of a metric defines when it will be reset. There are th
...
 ### Naming things
 
-Choose the category and names of metrics to be as specific as possible.
+Metric identifiers (the combination of a metric's category and name) must be unique across all metrics that are sent by an application.
+This includes not only the metrics defined in the app's `metrics.yaml`, but the `metrics.yaml` of any Glean-using library that that application uses, including Glean itself.
+Therefore, care should be taken to name things specifically enough so as to avoid namespace collisions.
+In practice, this generally involves thinking carefully about the `category` of the metric, more than the `name`.
+
+> **Note:** Duplicate metric identifiers are not currently detected at build time.
+However, the [probe_scraper](https://github.com/mozilla/probe-scraper) process, which runs nightly, will detect this case and e-mail the `notification_emails` associated with the given metrics.
```
Comment (Dexterp37) [inline docs/user/adding-new-metrics.md]: Do we have a bug for detecting at build time? if so, can we link it here ?

Comment (mdboom) [reply docs/user/adding-new-metrics.md]: No -- I just made a bug and linked to it here.

File: docs/user/adding-new-metrics.md [hunk 2]
```diff
@@ -70,7 +70,15 @@ The `lifetime` parameter of a metric defines when it will be reset. There are th
 
 ### Naming things
 
-Choose the category and names of metrics to be as specific as possible.
+Metric identifiers (the combination of a metric's category and name) must be unique across all metrics that are sent by an application.
+This includes not only the metrics defined in the app's `metrics.yaml`, but the `metrics.yaml` of any Glean-using library that that application uses, including Glean itself.
```
Comment (badboy) [inline docs/user/adding-new-metrics.md]: I always find "that that" weird to read. Any change to rephrase this? (I'm ok if not, don't spend too much time on this)

File: docs/user/adding-new-metrics.md (hunk 1 above)
Comment (badboy) [reply docs/user/adding-new-metrics.md]: The link in here is still missing.

**Pull request merged**